### ISSUE TITLE
[samcoupe][bug][getch] kbhit()/getch() freezes the computer

### ISSUE THREAD

**Issue by Fabrizio-Caruso** [z88dk/z88dk#1791]
I am using `getch()` and `kbhit()` to wait for a key.
The code below works on all targets across different dev-kits but freezes the samcoupe ramdonly (maybe when a key is pressed to long or too many times too rapidly):

```
#include<conio.h>
#define cgetc() getch()

void wait_for_key(void) {
    while(kbhit())
        (void) cgetc();
    while(!kbhit()) { }; 
    (void) cgetc();
}

int main(void)
{
    clrscr();
    while(1) {
        printf("press a key\n");
        wait_for_key();
    }
    return 0;
}
```

I compile with:
```
samcoupe_wait_for_key:
	$(Z88DK_PATH)$(MYZ88DK) +sam $(SCCZ80_OPTS) -O3 \
	-vn  \
	-create-app -o $(BUILD_PATH)/FULL_samcoupe.bin	\
	$(TEST_PATH)/wait_for_key.c
```
	
The same issue can be seen in my Cross-Lib demo "tiles", which is built with:
`make samcoupe -f Makefile.tiles`

**Comment by suborb** [comment]
This is an annoying side effect of #1756 - undoing the change there makes this snippet work correctly.

**Comment by suborb** [comment]
Ignore the red X above: the azure pipeline is playing up.

Anyway, I've fixed #1756 a different way and it keeps the behaviour of the fix for that issue + sorts this particular issue out - let me know how it goes for you.

**Comment by Fabrizio-Caruso** [comment]
It is fixed for me. Thanks!
